src/main/java/seedu/address/logic/commands/NearbyCommandUtil.java

```diff
@@ -0,0 +1,375 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import seedu.address.commons.core.index.Index;
+
+/**
+ * Contains utility methods used for Nearby Command
+ */
+public class NearbyCommandUtil {
+    private static HashMap<Integer, String> postalSectorInfo = new HashMap<>();
+
+    static {
+        initialize();
+    }
+
+    /**
+     * Checks if given {@code postalSector} is a valid postal sector.
+     *
+     * @param postalSector Used to find postal sector
+     * @return whether given {@code postalSector} is a valid postal sector
+     */
+    public static boolean isValidPostalSector(Index postalSector) {
+        requireNonNull(postalSector);
+        int givenPostalSector = postalSector.getOneBased();
+        return postalSectorInfo.containsKey(givenPostalSector);
+    }
+
+    /**
+     * Obtains the general location for a given {@code postalSector}.
+     *
+     * @param postalSector used to find general location
+     * @return {@code Optional<String>} representing the general location of given postal sector
+     */
+    public static Optional<String> getGeneralLocation(Index postalSector) {
+        requireNonNull(postalSector);
+        int givenPostalSector = postalSector.getOneBased();
+        return Optional.ofNullable(postalSectorInfo.get(givenPostalSector));
+    }
+
+    /**
+     * Obtain a list of postal sectors with the same general location.
+     *
+     * @param location used to find all matching postal sectors
+     * @return {@code List<String>} of all postal sectors with same location
+     */
+    public static List<String> sameGeneralLocation(String location) {
+        List<String> output = new ArrayList<>();
+        for (Map.Entry<Integer, String> entry : postalSectorInfo.entrySet()) {
+            if (entry.getValue().equals(location)) {
+                Integer sector = entry.getKey();
+                String sectorStr = "S" + ((sector < 10) ? ("0" + sector)
+                        : sector);
+                output.add(sectorStr);
+            }
+        }
+        return output;
+    }
+
+    /**
+     * Add postal sector information to HashMap {@code postalSectorInfo}.
+     */
+    private static void initialize() {
+        addSector1();
+        addSector7();
+        addSector14();
+        addSector9();
+        addSector11();
+        addSector17();
+        addSector18();
+        addSector20();
+        addSector22();
+        addSector24();
+        addSector28();
+        addSector31();
+        addSector34();
+        addSector38();
+        addSector42();
+        addSector46();
+        addSector49();
+        addSector51();
+        addSector53();
+        addSector56();
```

Comment:
Perhaps the naming could be more descriptive as to how many sectors are being added in each command.

Reply:
I think I should rename the methods to addLocation instead. Since the postal sectors share a common general location. Thanks for the feedback 👍